Common/Orders/OptionExercise/DefaultExerciseModel.cs

```diff
@@ -32,52 +32,56 @@ public class DefaultExerciseModel : IOptionExerciseModel
-                    (option.Symbol.ID.OptionRight == OptionRight.Call ? OrderDirection.Buy : OrderDirection.Sell);
 
-            var addUnderlyingEvent = new OrderEvent(order.Id,
-                            underlying.Symbol,
-                            utcTime,
-                            OrderStatus.Filled,
-                            exerciseDirection,
-                            exercisePrice,
-                            exerciseQuantity,
-                            OrderFee.Zero,
-                            "Option Exercise/Assignment");
+            var isShort = order.Quantity < 0;
+            var utcTime = option.LocalTime.ConvertToUtc(option.Exchange.TimeZone);
+
+            var inTheMoney = option.IsAutoExercised(underlying.Close);
 
-            var optionRemoveEvent = new OrderEvent(order.Id,
-                            option.Symbol,
-                            utcTime,
-                            OrderStatus.Filled,
-                            assignment ? OrderDirection.Buy : OrderDirection.Sell,
-                            0.0m,
-                            -optionQuantity,
-                            OrderFee.Zero,
-                            "Adjusting(or removing) the exercised/assigned option");
+            // we're assigned only if we get exercised against, meaning we wrote the option and it was exercised by the buyer
+            var isAssignment = inTheMoney && isShort;
 
-            if (optionRemoveEvent.FillQuantity > 0)
+            var optionRemoveEvent = new OrderEvent(
+                order.Id,
+                option.Symbol,
+                utcTime,
+                OrderStatus.Filled,
+                isShort ? OrderDirection.Buy : OrderDirection.Sell,
+                0.0m,
+                -order.Quantity,
+                OrderFee.Zero,
+                // note whether or not we expired OTM/ITM and whether we were assigned or exercised
+                inTheMoney ? isAssignment ? "Automatic Assignment" : "Automatic Exercise" : "OTM"
+            )
             {
-                optionRemoveEvent.IsAssignment = true;
-            }
+                IsAssignment = isAssignment
+            };
+
+            yield return optionRemoveEvent;
 
-            if (option.ExerciseSettlement == SettlementType.PhysicalDelivery &&
-                option.IsAutoExercised(underlying.Close))
+            if (inTheMoney && option.ExerciseSettlement == SettlementType.PhysicalDelivery)
             {
-                return new[] { optionRemoveEvent, addUnderlyingEvent };
-            }
+                var right = option.Symbol.ID.OptionRight;
 
-            return new[] { optionRemoveEvent };
-        }
+                // TODO : Why doesn't this method take into account the directionality of the quantity like other quantity properties/methods
+                var fillQuantity = option.GetExerciseQuantity(order.Quantity);
```

Comment:
> // TODO : Why doesn't this method take into account the directionality of the quantity like other quantity properties/methods

You are referring to the required `right == OptionRight.Call ? fillQuantity : -fillQuantity`? All consumers of `GetExerciseQuantity` seem to be doing the same operation except for `OptionExerciseOrder` that seems like a bug? 🤔 Agree it kinda feels it should be internal to the `GetExerciseQuantity`

Reply:
my thoughts exactly -- throughout lean we use `Quantity` to mean the directional quantity and `AbsoluteQuantity` to mean the magnitude of the quantity. I'll put an additional commit in here refactoring this method to follow this convention.